cookbook/mass_edition/register_a_new_mass_edit_action.rst

```diff
@@ -2,55 +2,111 @@ How to Register a New Mass Edit Action on Products
+It also comes with a flexible way to define your own mass edit action
 on selected products.
 
-Creating a MassEditAction
--------------------------
-The first step is to create a new class that implements ``MassEditActionInterface`` or extends
-``ProductMassEditOperation`` or ``FamilyMassEditOperation`` (given on which grid you want to apply the operation):
 
-.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
-   :language: php
-   :prepend: # /src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
-   :linenos:
+Prerequisite
+------------
+The mass edit action uses the `BatchBundle
+<https://github.com/akeneo/BatchBundle/>`_ in order to run mass edit in background. Readers and Writers are already
+created so in this cookbook we will focus on how to create a Mass Edit Action and create a Processor.
+For more information on how to create a Job, Reader, Processor, or Writer please see 'create-specific-connector'
 
-.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
-  :language: php
-  :prepend: # /src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
-  :linenos:
+Work with a custom Acme bundle
+------------------------------
+Your custom Mass Edit actions have to be in a custom Acme bundle which inherit our ``EnrichBundle``.
+We advise you to create this custom bundle with the Symfony command:
+
+.. code-block:: bash
+
+  php app/console generate:bundle
+
+Once your bundle is created, we must inform Symfony it inherits our Bundle:
+
+.. code-block:: php
+
+    # /src/Acme/Bundle/EnrichBundle/AcmeEnrichBundle.php
+    <?php
 
+    namespace Acme\Bundle\EnrichBundle;
 
-This class will contain all the information about the operation to run and the form type which is used to configure it.
+    use Symfony\Component\HttpKernel\Bundle\Bundle;
 
+    class AcmeEnrichBundle extends Bundle
+    {
+        public function getParent()
+        {
+            return 'PimEnrichBundle';
+        }
+    }
 
-Registering the MassEditAction
+Phase 1: Create the Operation
 ------------------------------
 
-After the class is created, you must register it as a service in the DIC with the pim_catalog.mass_edit_action tag:
+.. tip::
 
-By default, the operation will be available for the product grid.
-It is possible to apply the operation on the family grid though.
+    **Operations** are designed to build and transport the configuration (*eventually via a form*) that will be send to the background job.
+    No item is updated from here!
 
-.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/Resources/config/services.yml
-   :language: yaml
-   :prepend: # /src/Acme/Bundle/EnrichBundle/Resources/config/services.yml
+The first step is to create a new class in the Operation folder that extends ``AbstractMassEditOperation`` and declare this new class as a service in the mass_action.yml file
```

Comment:
mass_actions.yml !
Be really carefull with that kind of errors. It is the main problem in the current cookbook. File path or name are not good and the cookbook can't be done.